asserts/tool/sign_test.go

```diff
@@ -0,0 +1,331 @@
+	c.Assert(err, IsNil)
+
+	now := time.Now()
+	// good enough as a handle as is used by Sign
+	mockAccKey := "type: account-key\n" +
+		"authority-id: canonical\n" +
+		"account-id: user-id1\n" +
+		"public-key-id: " + s.testKeyID + "\n" +
+		"public-key-fingerprint: " + testKey.PublicKey().Fingerprint() + "\n" +
+		"since: " + now.Format(time.RFC3339) + "\n" +
+		"until: " + now.AddDate(1, 0, 0).Format(time.RFC3339) + "\n" +
+		fmt.Sprintf("body-length: %v", len(pubKeyEncoded)) + "\n\n" +
+		string(pubKeyEncoded) + "\n\n" +
+		"openpgp c2ln"
+
+	s.accKey = []byte(mockAccKey)
+}
+
+const (
+	flatModelYaml = `series: "16"
+brand-id: user-id1
+model: baz-3000
+core: core
+architecture: amd64
+gadget: brand-gadget
+kernel: baz-linux
+store: brand-store
+allowed-modes:
+required-snaps: [foo,bar]
+class: fixed
+extra-flag: true
+timestamp: 2015-11-25T20:00:00Z
+`
+	nestedModelYaml = `headers:
+  series: "16"
+  brand-id: user-id1
+  model: baz-3000
+  core: core
+  architecture: amd64
+  gadget: brand-gadget
+  kernel: baz-linux
+  store: brand-store
+  allowed-modes:
+  required-snaps: [foo,bar]
+  class: fixed
+  extra-flag: true
+  timestamp: 2015-11-25T20:00:00Z
+`
+)
+
+func expectedModelHeaders() map[string]string {
+	return map[string]string{
+		"type":           "model",
+		"authority-id":   "user-id1",
+		"series":         "16",
+		"brand-id":       "user-id1",
+		"model":          "baz-3000",
+		"allowed-modes":  "",
+		"architecture":   "amd64",
+		"class":          "fixed",
+		"extra-flag":     "yes",
+		"gadget":         "brand-gadget",
+		"kernel":         "baz-linux",
+		"core":           "core",
+		"store":          "brand-store",
+		"required-snaps": "foo,bar",
+		"timestamp":      "2015-11-25T20:00:00Z",
+	}
+}
+
+func (s *signSuite) TestSignKeyIDFlatYAML(c *C) {
+	req := tool.SignRequest{
+		KeyID:       s.testKeyID,
+		AuthorityID: "user-id1",
+
+		AssertionType:      "model",
+		StatementMediaType: "application/x-yaml",
+		Statement:          []byte(flatModelYaml),
+	}
+
+	assertText, err := tool.Sign(&req, s.keypairMgr)
+	c.Assert(err, IsNil)
+
+	a, err := asserts.Decode(assertText)
+	c.Assert(err, IsNil)
+
+	c.Check(a.Type(), Equals, asserts.ModelType)
+	c.Check(a.Revision(), Equals, 0)
+	c.Check(a.Headers(), DeepEquals, expectedModelHeaders())
+	c.Check(a.Body(), IsNil)
+}
+
+func (s *signSuite) TestSignKeyIDNestedYAML(c *C) {
+	req := tool.SignRequest{
+		KeyID:       s.testKeyID,
+		AuthorityID: "user-id1",
+
+		AssertionType:      "model",
+		StatementMediaType: "application/x-yaml",
+		Statement:          []byte(nestedModelYaml),
+	}
+
+	assertText, err := tool.Sign(&req, s.keypairMgr)
+	c.Assert(err, IsNil)
+
+	a, err := asserts.Decode(assertText)
+	c.Assert(err, IsNil)
+
+	c.Check(a.Type(), Equals, asserts.ModelType)
+	c.Check(a.Revision(), Equals, 0)
+	c.Check(a.Headers(), DeepEquals, expectedModelHeaders())
+	c.Check(a.Body(), IsNil)
+}
+
+func (s *signSuite) TestSignKeyIDNestedYAMLWithBodyAndRevision(c *C) {
```

Comment:
The "AndRevision" part of this (and TestSignKeyIDNestedJSONWithBodyAndRevision below) feels like it should be a separate test, e.g. TestRequestRevisionOverridesHeaderReevision or similar.